src/logworker.cpp

```diff
@@ -113,21 +113,22 @@ namespace g3 {
    }
 
    void LogWorker::addWrappedSink(std::shared_ptr<g3::internal::SinkWrapper> sink) {
-      auto bg_addsink_call = [this, &sink] {_impl._sinks.push_back(sink);};
-      auto token_done = g3::spawn_task(bg_addsink_call, _impl._bg.get());
-      token_done.wait();
+      auto bg_addsink_call = [this, sink] {_impl._sinks.push_back(sink);};
+      g3::spawn_task(bg_addsink_call, _impl._bg.get());
```

Comment:
Ah. I just thought of something.  I remembered why I had the `wait()` here.  It was just to be explicit about what's going on and I did not want to return a `std::future<>` (and at the time I wanted to know when the sink was in place)

So even if you don't do the `wait()` here. The actual result of the code hasn't changed.   The `g3::spawn_task(...)` returns a future. Since you are not capturing the future it falls out of scope and so it has the same effect as calling `wait()`.  

I believe you can just use the `send(...)` function instead if you want to avoid waiting for the `std::future`